tests/test_views.py

```diff
@@ -0,0 +1,60 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self) -> None:
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="admin",
+            password="test12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self) -> None:
+        Manufacturer.objects.create(name="Test_one")
+        Manufacturer.objects.create(name="Test_two")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(
+            response, "taxi/manufacturer_list.html"
+        )
+
+
+class PrivateDriverTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="admin_two",
+            password="pass12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_create_driver(self):
+        form_data = {
+            "username": "A_B123",
+            "password1": "anton_best123",
+            "password2": "anton_best123",
+            "first_name": "Anton",
+            "last_name": "John",
+            "license_number": "AA0001AO"
+        }
+        self.client.post(reverse("taxi:driver-create"), data=form_data)
```

Comment:
Consider checking the response status code after the POST request to ensure the form submission was successful. This can help catch any potential validation errors or issues with the form processing.